### PR TITLE
refactor: createPR function (#4996)

### PR DESCRIPTION
**Pull request by souravdasslg**
1. New Interface is created named CreatePRConfig
2. Every function call is refactored.
3. Updated tests and snapshots accordingly.

#4996

### CODE REVIEW

**Review by JamieMagee — DISMISSED** [review]

File: lib/platform/bitbucket-server/index.ts [hunk 1]
```diff
@@ -859,13 +860,13 @@ export async function ensureCommentRemoval(
...
-  title: string,
-  rawDescription: string,
-  _labels?: string[] | null,
-  useDefaultBranch?: boolean
-): Promise<Pr> {
+export async function createPr({
+  branchName,
+  prTitle: title,
+  prBody: rawDescription,
+  labels: _labels,
```
Comment (ViceIce) [inline lib/platform/bitbucket-server/index.ts]:
```suggestion
```

File: lib/platform/bitbucket/index.ts [hunk 2]
```diff
@@ -667,13 +668,13 @@ export function ensureCommentRemoval(
...
-  title: string,
-  description: string,
-  _labels?: string[],
-  useDefaultBranch = true
-): Promise<Pr> {
+export async function createPr({
+  branchName,
+  prTitle: title,
+  prBody: description,
+  labels: _labels,
```
Comment (ViceIce) [inline lib/platform/bitbucket/index.ts]:
```suggestion
```

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/platform/bitbucket-server/index.ts (hunk 1 above)
Comment (souravdasslg) [reply lib/platform/bitbucket-server/index.ts]: Removed as suggested.

File: lib/platform/bitbucket/index.ts (hunk 2 above)
Comment (souravdasslg) [reply lib/platform/bitbucket/index.ts]: Removed as suggested.

**Review by viceice — APPROVED** [review]

**Review by viceice — CHANGES_REQUESTED** [review]
needs deconflicting

**Review by viceice — APPROVED** [review]

**Review by viceice — CHANGES_REQUESTED** [review]
needs deconflicting again 😅

**Comment by souravdasslg** [comment]
> needs deconflicting again 😅

Done. I was waiting for this :stuck_out_tongue:

**Review by viceice — APPROVED** [review]

**Comment by ViceIce** [comment]
@rarkins Ready for merge

**Pull request merged**